js/stex.js

```diff
@@ -0,0 +1,803 @@
+            this.safeFloat (ohlcv, 'open'),
+            this.safeFloat (ohlcv, 'high'),
+            this.safeFloat (ohlcv, 'low'),
+            this.safeFloat (ohlcv, 'close'),
+            this.safeFloat (ohlcv, 'volume'),
+        ];
+    }
+
+    async fetchOHLCV (symbol, timeframe = '1d', since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let market = this.market (symbol);
+        if (limit === undefined)
+            limit = 100;
+        if (since !== undefined)
+            params['since'] = since;
+        let response = await this.privatePostGrafic (this.extend ({
+            'pair': market['id'],
+            'interval': this.timeframes[timeframe],
+            'page': 1,
+            'count': limit,
+        }, params));
+        let data = this.getdatafromresponse (response);
+        return this.parseOHLCVs (this.getdatafromresponse (data), market, timeframe, since, limit);
+    }
+
+    async createDepositAddress (code, params = {}) {
+        let response = await this.privatePostGenerateWallets (this.extend ({
+            'currency': code,
+        }, params));
+        let data = this.getdatafromresponse (response);
+        let address = undefined;
+        let tag = undefined;
+        if (data !== undefined) {
+            let address = this.safeString (data, 'address');
+            this.checkAddress (address);
+            tag = this.safeString (data, 'msg');
+        }
+        return {
+            'currency': code,
+            'address': address,
+            'tag': tag,
+            'info': response,
+        };
+    }
+
+    async fetchDepositAddress (code, params = {}) {
+        let response = await this.privatePostGetInfo ();
+        let address = undefined;
+        let tag = undefined;
+        let data = this.getdatafromresponse (response);
+        if ('wallets_addresses' in data) {
+            if (code in data['wallets_addresses'])
+                address = data['wallets_addresses'][code];
+        }
+        if ('publick_key' in data) {
+            if (code in data['publick_key'])
+                tag = data['publick_key'][code];
+        }
+        this.checkAddress (address);
+        return {
+            'currency': code,
+            'address': address,
+            'tag': tag,
+            'info': address,
+        };
+    }
+
+    processTransactionHelper (data = undefined, txtype = undefined) {
+        let result = {};
+        let ids = Object.keys (data[txtype]);
+        for (let i = 0; i < ids.length; i++) {
+            let id = ids[i];
+            result.push (this.extend ({ 'id': id, 'type': txtype.toLowerCase () }, data[txtype][id]));
+        }
+        return result;
+    }
+
+    async fetchTransHistory (request, type = undefined) {
+        if (type === 'all') {
+            let response1 = await this.privatePostTransHistory (this.extend ({ 'status': 'FINISHED' }, request));
```

Comment:
this is not ok...

too many requests in one method (one request per method is the rule)

moreover can you drop `this.getdatafromresponse` and expand it everywhere so we are explicit in what is happening

Reply:
Yeah, its a bunch of calls.. But its the only way to get the complete history on Stex. At the moment it is only getting data from "FINISHED" call since type is undefined. I was planning on requiring a user supplied param or so in fetchTransaction to allow fetching complete data. (if (params['complete'] !== undefined) or so as an example.)

**Are we good with getting only information from finished transactions?** Personally i would like to have information from rest also, or at least have the ability to get them, but it totals in 9 calls which is a lot.

 There are also a couple other commands that includes at least 2 calls due to limitation in stex api.. For example in fetchOrder, it will first check ActiveOrders for match, if no match there it will then check  TradeHistory->pending/prossesing/finish/canceled, and return once it finds a match. Without these, it may return without finding a order in canceled state as an example.

Im open for suggestions and tips.

Reply:
> EDIT: It seems to be to do with transactions, in that case, their api sucks and I don't know how to deal with it. Check their api docs if they have an 'all' status or don't send the status at all.

According to they're api docs, it will only return finished tx's if no param is given, and only finished tx's if param 'ALL' is used, so yeah it sucks.

> status: an integer value which represent deposit or withdrawal status (optional, default: "FINISHED")
> If parameter operation is set to "ALL" parameter status will be set to " FINISHED".

Have not tried to set status param to ALL on tradehistory api yet, will give it a try tomorrow since they're docs dont specify it.